eglib/src/gstr.c

```diff
@@ -251,10 +258,16 @@ g_strerror (gint errnum)
 			error_messages [errnum] = g_strdup (buff);
 		if (buff != tmp_buff)
 			g_free (buff);
-#else
+#else /* HAVE_STRERROR_R_XSI */
+		buff = strerror_r (errnum, buff, buff_len);
```

Comment:
Usually, I trust the man pages - I guess that's a safe bet? In addition, it also makes a lot of sense that these strings are immutable; the custom string buffer will only be used for personalised messages.